atlp-rwanda/rca-barfoot-team3-bn
# feat: allow users to view all the requests they have sent

### What does this PR do?

This PR adds an endpoint for the users to view the requests they have sent through out the time.


### Description of Task to be completed?

It first adds a new model called Request.

which has one OneWayTrip and has many rooms, because you can book multiple rooms in many hotels.

and then configures the relations between the models

I also fixed some issues on the search like when there are empty results instead of throwing an error return an empty list


### How should this be manually tested?

Clone the application

Make sure you are logged in

Hit GET /api/v1/trip/requests


### Any background context you want to provide?
None


### What are the relevant pivotal tracker/Trello stories?

#184638006 - Integrate the requests table to the backend
#184638013 - Add Search Functionality 


### Screenshots (if appropriate)

<img width="1380" alt="image" src="https://user-images.githubusercontent.com/61510549/232733584-ed235515-cff1-4237-9edc-e20b849800ef.png">

<img width="1334" alt="image" src="https://user-images.githubusercontent.com/61510549/232733783-c2ed886b-4a56-42d8-90cc-2110a2d5c46b.png">

<img width="1307" alt="image" src="https://user-images.githubusercontent.com/61510549/232733918-9e0590c3-c469-4b30-b188-41daeacd3e2c.png">
